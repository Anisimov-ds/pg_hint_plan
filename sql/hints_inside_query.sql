LOAD 'pg_hint_plan';

explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';

set pg_hint_plan.hints_inside_query = on;
explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';

set pg_hint_plan.hints_inside_query = off;
explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';

set pg_hint_plan.hints_inside_query = on;
/*+ MergeJoin(t1 t2) */
explain (costs false)
select * from t1 join t2 on t1.val = t2.val where '/*+HashJoin(t1 t2)*/' <> '';

/*+ HashJoin(t1 t2) */
explain (costs false)
select * from t1 join t2 on t1.val = t2.val where '/*+MergeJoin(t1 t2)*/' <> '';
