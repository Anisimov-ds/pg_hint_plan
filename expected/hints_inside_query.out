LOAD 'pg_hint_plan';
explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';
              QUERY PLAN              
--------------------------------------
 Merge Join
   Merge Cond: (t1.id = t2.id)
   ->  Index Scan using t1_pkey on t1
   ->  Index Scan using t2_pkey on t2
(4 rows)

set pg_hint_plan.hints_inside_query = on;
explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';
          QUERY PLAN          
------------------------------
 Hash Join
   Hash Cond: (t1.id = t2.id)
   ->  Seq Scan on t1
   ->  Hash
         ->  Seq Scan on t2
(5 rows)

set pg_hint_plan.hints_inside_query = off;
explain (costs false)
select * from t1 join t2 on t1.id = t2.id where '/*+HashJoin(t1 t2)*/' <> '';
              QUERY PLAN              
--------------------------------------
 Merge Join
   Merge Cond: (t1.id = t2.id)
   ->  Index Scan using t1_pkey on t1
   ->  Index Scan using t2_pkey on t2
(4 rows)

set pg_hint_plan.hints_inside_query = on;
/*+ MergeJoin(t1 t2) */
explain (costs false)
select * from t1 join t2 on t1.val = t2.val where '/*+HashJoin(t1 t2)*/' <> '';
                QUERY PLAN                 
-------------------------------------------
 Merge Join
   Merge Cond: (t2.val = t1.val)
   ->  Index Scan using t2_val on t2
   ->  Materialize
         ->  Index Scan using t1_val on t1
(5 rows)

/*+ HashJoin(t1 t2) */
explain (costs false)
select * from t1 join t2 on t1.val = t2.val where '/*+MergeJoin(t1 t2)*/' <> '';
           QUERY PLAN           
--------------------------------
 Hash Join
   Hash Cond: (t2.val = t1.val)
   ->  Seq Scan on t2
   ->  Hash
         ->  Seq Scan on t1
(5 rows)

